src/main/java/de/relimit/commons/markdown/blockelement/heading/Heading.java

```diff
@@ -85,13 +95,32 @@ public void setLevel(int level) {
 		invalidateSerialized();
 	}
 
+	/**
+	 * @deprecated Use {@link #getHeadingStyle()} and check if equals
+	 *             {@link HeadingStyle#SETEXT} instead
+	 * @return
+	 */
+	@Deprecated
 	public boolean isUnderlineStyle() {
-		return underlineStyle;
+		return HeadingStyle.SETEXT == headingStyle;
 	}
 
+	/**
+	 * @deprecated Use {@link #setHeadingStyle(HeadingStyle)} instead.
+	 * @param underlineStyle
+	 */
+	@Deprecated
 	public void setUnderlineStyle(boolean underlineStyle) {
-		this.underlineStyle = underlineStyle;
+		setHeadingStyle(underlineStyle ? HeadingStyle.SETEXT : HeadingStyle.ATX);
+	}
+
+	public void setHeadingStyle(HeadingStyle headingStyle) {
+		this.headingStyle = headingStyle;
 		invalidateSerialized();
 	}
 
+	public HeadingStyle getHeadingStyle() {
+		return headingStyle != null ? DEFAULT_HEADING_STYLE : headingStyle;
```

Comment:
The logic seems wrong here. Should it not be
return headingStyle != null ? headingStyle : DEFAULT_HEADING_STYLE;

Which means: If the member `headingStyle` is not null, so is assigned, get this `headingStyle`, else use `DEFAULT_HEADING_STYLE

Reply:
Ouch. Nice catch! 👍 I fixed it and now use `Optional` to make it clearer what is going on.

Reply:
These changes look fine now